src/Interpreters/evaluateConstantExpression.cpp

```diff
@@ -42,6 +42,21 @@ static std::pair<Field, std::shared_ptr<const IDataType>> getFieldAndDataTypeFro
     return {res, type};
 }
 
+static String getColumnName(const IAST & ast)
+{
+    try
+    {
+        return ast.getColumnName();
+    }
+    catch (Exception & e)
+    {
+        if (e.code() == ErrorCodes::LOGICAL_ERROR && startsWith(e.message(), "Trying to get name of not a column"))
```

Comment:
It will not work because we call `abort` in the `Exception` ctor before it was thrown. It's impossible to catch `LOGICAL_ERROR`